File: algorithms/fisher_yates/java/fisher_yates.java
import java.util.Random;

public class fisher_yates{

	public static void main(String[] args) {
			int[] arr = {1,2,3,4,5,6,7,8,9};

			System.out.println("Array before shuffling:");
			for(int i: arr) System.out.print(i + ", ");

			System.out.println("\n\nShuffling...");
			arr = shuffle(arr);

			System.out.println("\nArray after shuffling:");
			for(int i: arr) System.out.print(i + ", ");
	}


	
	public static int[] shuffle(int[] arr) {
		int switchIndex = arr.length - 1;
		int roll; 
		int temp;

		Random rng = new Random();

		while(switchIndex != 0) {
			roll = rng.nextInt(switchIndex);
			temp = arr[roll];
			arr[roll] = arr[switchIndex];
			arr[switchIndex] = temp;
			switchIndex--;
		}

		return arr;
	}
}

File: algorithms/fisher_yates/python/fisher_yates.py
import random

def fisher_yates(arr):
	switchIndex = len(arr) -1

	while(switchIndex != 0):
		roll = random.randrange(switchIndex)
		temp = arr[roll]
		arr[roll] = arr[switchIndex]
		arr[switchIndex] = temp
		switchIndex = switchIndex - 1
	return arr

print('Array before shuffling')
arr = [1,2,3,4,5,6,7,8,9]
print(arr)

print('\nShuffling...')
arr = fisher_yates(arr)

print('\nArray after shuffling')
print(arr)